docusaurus/docs/iOS/guides/push-notifications.md

```diff
@@ -1,3 +1,179 @@
+2. Go to **Keychain Access**
+3. At the top left, select **Keychains > Login**
+4. Then, at the bottom left, select **Category > Certificates**
+5. Select the certificate you've created in the previous step. It should look like `Apple Development IOS Push Services: YOUR_APP_NAME` and expand it to see the private key(it should be named after the Name you provided when creating the `Certificate Signing Request`
+6. Right-click the private key and click on **Export**. In the **File** format section select **Personal Information Exchange (.p12)** and save the file on your hard drive
+  
+## Step 4. Uploading the Certificate to Stream Chat
+
+Certificate uploading process is the same regardless of authentication method and was [covered in detail above](#step-4-uploading-the-key-credentials-to-stream-chat).
+
+Essentially you just need to run a setup wizzard:
+
+    stream chat:push:apn
+
+
+## Testing Push Notifications setup
+
+First of all make sure you have at least one [device associated](#managing-users-for-testing-purposes).
+
+If the device you want to test notifications on is already added to the list, you can now test pushes:
+
+    stream chat:push:test
+
+This will do several things for you:
+
+1. Pick a random message from a channel that this user is part of
+2. Use the notification templates configured for your push providers to render the payload using this message
+3. Send this payload to all of the user's devices
+   
+## Managing users for testing purposes
+
+For testing purposes you can manage users using Stream CLI:
+
+### Adding users
+    stream chat:push:device:add
+
+
+### Removing users
+    stream chat:push:device:delete
+
+### Getting the list of all devices registered for pushes
+    stream chat:push:device:get
+
+## Managing users in client code
+
+### Adding users
+
+```swift
+func application(
+    _ application: UIApplication,
+    didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data
+) {
+    chatClient.currentUserController().addDevice(token: deviceToken)
+    // or
+    chatClient.currentUserController().addDevice(token: deviceToken) { error in
+        if let error = error {
+            // handle error
+            print(error)
+        }
+    }
+}
+```
+
+### Removing users
+
+```swift
+let deviceId = chatClient.currentUserController().currentUser!.devices.last!.id
+
+chatClient.currentUserController().removeDevice(id: deviceId)
+// or
+chatClient.currentUserController().removeDevice(id: deviceId) { error in
+    if let error = error {
+        // handle error
+        print(error)
+    }
+}
+```
+
+## Push Delivery Logic
+
+Only new messages are pushed to mobile devices, all other chat events are only send to WebSocket clients and webhook endpoints if configured.
+
+Push message delivery follows the following logic:
+
+* Only channel members can receive push messages
+* Members that are currently online do not receive push messages
```

Comment:
"currently online" -> "watching the channel", and link to "Watching" documentation (whichever that is...)

Reply:
Do you mean this page? https://getstream.io/chat/docs/ios-swift/watch_channel/?language=swift

Reply:
i mean our documentation "Working with Watchers", but the one you linked would be better since we don't have "Working with Watchers" yet